src/fight/computation_tools.rs

```diff
@@ -0,0 +1,473 @@
+    attacking_stats: &model::Stats,
+    defending_stats: &model::Stats,
+) -> f64 {
+    let minimum_to_hit: usize = compute_roll_to_hit(
+        attacking_stats.get_offensive(),
+        defending_stats.get_defense(),
+    );
+    let minimum_to_wound: usize = compute_roll_to_wound(
+        attacking_stats.get_strength(),
+        defending_stats.get_resilience(),
+    );
+    let hit_probability: f64 = (global_values::DEFAULT_DICE - (minimum_to_hit as f64) + 1.0_f64)
+        / (global_values::DEFAULT_DICE);
+    let wound_probability: f64 = (global_values::DEFAULT_DICE - (minimum_to_wound as f64)
+        + 1.0_f64)
+        / (global_values::DEFAULT_DICE);
+    match hit_probability * wound_probability {
+        x if x <= 0.0 => 0.0,
+        y => y,
+    }
+}
+
+/// Return the fastest Model between 2 Model
+/// If the first Model is faster than the second Model, return 1
+/// If the second Model is faster than the first Model, return 2
+/// If the Models have the same speed, return 0
+///
+/// ## Parameters
+/// attacking_stats (&model::Stats): The attacker stats
+///
+/// defending_stats (&model::Stats): The defender stats
+///
+/// ## Return
+/// u8: The fastest model
+///
+/// 0 if the two units have the same speed
+///
+/// 1 if the attacking is the fastest
+///
+/// 2 if the defending is the fastest)
+pub fn find_the_fastest(attacking_stats: &model::Stats, defending_stats: &model::Stats) -> u8 {
+    let attacking_model_agility: usize = attacking_stats.get_agility() + 1;
+    let defending_model_agility: usize = defending_stats.get_agility();
+
+    match attacking_model_agility.cmp(&defending_model_agility) {
+        std::cmp::Ordering::Greater => 1,
+        std::cmp::Ordering::Less => 2,
+        std::cmp::Ordering::Equal => 0,
+    }
+}
+
+/// Compute the probability for a model to protect itself from another model damages
+///
+/// ## Parameters
+/// (&model::Stats) defending_stats: The defender stats
+///
+/// (&model::Stats) attacking_stats: The attacker stats
+///
+/// ## Return
+/// f64: The probability to save a damage dealt by attacker
+fn compute_save_probability(defending_stats: &model::Stats, attacking_stats: &model::Stats) -> f64 {
+    let armour_save: usize = match global_values::ARMOUR_SAVE_THRESHOLD as isize
+        - (defending_stats.get_armour() as isize
+            - attacking_stats.get_armour_penetration() as isize)
+    {
+        x if x < 0 => 0usize,
+        y => y as usize,
+    };
+    let save_proba: f64 = match (global_values::DEFAULT_DICE - armour_save as f64 + 1.0)
+        / global_values::DEFAULT_DICE
+    {
+        x if x <= 0.0 => 0.0,
+        x if x > 1.0 => 1.0,
+        y => y,
+    };
+    let aegis_proba: f64 = match (global_values::DEFAULT_DICE - defending_stats.get_aegis() as f64
+        + 1.0)
+        / global_values::DEFAULT_DICE
+    {
+        x if x <= 0.0 => 0.0,
+        x if x > 1.0 => 1.0,
+        y => y,
+    };
+    if defending_stats.get_aegis() == 0 {
```

Comment:
2 calls to get_aegis should be reunited in a variable

Reply:
noted :+1: